src/core/model/Layer.cpp

```diff
@@ -144,7 +131,11 @@ auto Layer::isVisible() const -> bool { return visible; }
  */
 void Layer::setVisible(bool visible) { this->visible = visible; }
 
-auto Layer::getElements() const -> const std::vector<Element*>& { return this->elements; }
+auto Layer::getElements() const -> std::vector<Element*> {
+    std::vector<Element*> res{this->elements.size(), nullptr};
+    std::transform(this->elements.begin(), this->elements.end(), res.begin(), [](auto const& e) { return e.get(); });
+    return res;
+}
```

Comment:
I think it's never called for drawing, but it is called at some places. I previously returned a new Element vector, because it was required for Element Containers, to return an Element vector by value.
There are also many cases, where the returned vector is looped over, to remove an element from the Layer. 
In all those cases, we must do a copy before.

But I might move all copies to the place we actually need that copy.